src/levanter/main/train_lm.py

```diff
@@ -19,12 +19,29 @@
 from levanter.models.lm_model import LmConfig
 from levanter.optim import AdamConfig, OptimizerConfig
 from levanter.trainer import Trainer, TrainerConfig
+from levanter.types import ComputeLossFunction, M, X
 from levanter.utils.jax_utils import parameter_count
 
 
 logger = logging.getLogger(__name__)
 
 
+class ModuleComputeZLoss(ComputeLossFunction[M, X]):
```

Comment:
i still don't like this but I think I can't really articulate what I want. i'm gonna push a change to my fork